### PR TITLE
fixed issue1788: removed strict check

### PR DESCRIPTION
**Pull request by atorian**
Issue: https://github.com/Codeception/Codeception/issues/1788
i've investigated a bit more and found out that it would be better be go with php way - make it array_diff does http://php.net/manual/ru/function.array-diff.php

    Note:

    Two elements are considered equal if and only if (string) $elem1 === (string) $elem2. In words: when the string representation is the same.

but with current code it will look like simple non-strict check.
So i've changed that condition and applied tests, to proof new behaviour. 
Please verify.

### CODE REVIEW

**Comment by Naktibalda** [comment]
It looks good to me.

**Comment by Naktibalda** [comment]
Actually the check is way too loose now.

1. ==  casts string to integer if it starts with 0, examples: http://3v4l.org/tT4l8
2. Values of different types have the same string representation - false == null == "", true == "1" == 1
  false, null and true values are very easy to distinguish in both JSON and PHP, I think that we should not use a loose comparison for them,

**Comment by atorian** [comment]
The issue could be partially fixed with json_numeric_check flag passed to json_encode in the app. But this is not always possible, or requires to many effort.
But it won't fix issues with dynamic expected values. For example if value was grabbed from db or file - it will have type === string.
I'll investigate more.

**Comment by Naktibalda** [comment]
Since this issue is specific to numbers, it could be solved by casting all integers and floats to string and then comparing strictly, right?

**Comment by atorian** [comment]
i believe so.
I'll do this today.

**Comment by atorian** [comment]
please verify

File: src/Codeception/Util/JsonArray.php [hunk 1]
```diff
@@ -212,4 +211,17 @@ private function arrayToXml(\DOMDocument $doc, \DOMNode $node, $array)
             }
         }
     }
+
+    private function isEqualValue($val1, $val2)
+    {
+        if (is_numeric($val1)) {
```
Comment (sergeyklay) [inline src/Codeception/Util/JsonArray.php]: mb [is_scalar](http://php.net/manual/en/function.is-scalar.php) better way?

Comment (Naktibalda) [inline src/Codeception/Util/JsonArray.php]: It would cast boolean values to strings, which is not necessary.

**Pull request merged**